### PR TITLE
Fixes flaky UPE E2E tests

### LINKED ISSUE #6322: Fix flaky UPE tests
### Describe the bug

Frequently yet intermittently, the two tests "WC - latest | upe - shopper" and "WC - latest | UPESplit - shopper" keep failing in GH actions for WCPay PRs. 

There are some problems with the new UPE deferred intent E2E tests, but the UPE E2E tests also appear to have an issue on [this line](https://github.com/Automattic/woocommerce-payments/blob/develop/tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js#L23) where SEPA is enabled, more specifically causing an error on [this line](https://github.com/Automattic/woocommerce-payments/blob/develop/tests/e2e/utils/flows.js#L479) where the node/button in question appears not to exist.

### To Reproduce

1. Go to any recent PR and check out the failing tests. 

### Actual behavior

[Example failed jobs](https://github.com/Automattic/woocommerce-payments/actions/runs/5002564246/jobs/8962671002?pr=6028).

UPE tests fail...
```bash
FAIL tests/e2e/specs/upe/shopper/shopper-upe-enabled-all-flows.spec.js (77.377 s)
  Enabled UPE
    Shortcode checkout
      ✕ should save the card (18494 ms)
      ✕ should process a payment with the saved card (10919 ms)
      ✕ should delete the card (3199 ms)
    My Account
      ✕ should add the card as a new payment method (8751 ms)
      ✕ should be able to delete the card (2085 ms)

  ● Enabled UPE › Shortcode checkout › should save the card

    Node is either not visible or not an HTMLElement

      477 | 			);
      478 | 			if ( tourKitDismissButton ) {
    > 479 | 				await tourKitDismissButton.click();
          | 				^
      480 | 			}
      481 |
      482 | 			await page.$eval( paymentMethod, ( method ) => method.click() );

      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:221:13)
          at runMicrotasks (<anonymous>)
      at ElementHandle.click (node_modules/puppeteer/lib/JSHandle.js:282:20)
      at Object.enablePaymentMethod (tests/e2e/utils/flows.js:479:5)
      at Object.<anonymous> (tests/e2e/specs/upe/shopper/shopper-upe-enabled-all-flows.spec.js:23:3)
        -- ASYNC --
      at ElementHandle.click (node_modules/puppeteer/lib/helper.js:111:15)
      at Object.enablePaymentMethod (tests/e2e/utils/flows.js:479:32)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (tests/e2e/specs/upe/shopper/shopper-upe-enabled-all-flows.spec.js:23:3)
```

UPE split tests fail...
```bash
  ● Enabled enhanced UPE › should successfully place order with the default card

    Node is either not visible or not an HTMLElement

      474 | 			);
      475 | 			if ( tourKitDismissButton ) {
    > 476 | 				await tourKitDismissButton.click();
          | 				^
      477 | 			}
      478 |
      479 | 			await page.$eval( paymentMethod, ( method ) => method.click() );

      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:221:13)
          at runMicrotasks (<anonymous>)
      at ElementHandle.click (node_modules/puppeteer/lib/JSHandle.js:282:20)
      at Object.enablePaymentMethod (tests/e2e/utils/flows.js:476:5)
      at Object.<anonymous> (tests/e2e/specs/upe-split/shopper/shopper-deferred-intent-creation-upe-enabled.spec.js:31:3)
        -- ASYNC --
      at ElementHandle.click (node_modules/puppeteer/lib/helper.js:111:15)
      at Object.enablePaymentMethod (tests/e2e/utils/flows.js:476:32)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (tests/e2e/specs/upe-split/shopper/shopper-deferred-intent-creation-upe-enabled.spec.js:31:3)
```

### Screenshots

See above.

### Expected behavior

Tests should not fail arbitrarily.

### Desktop (please complete the following information):

N/A

### Smartphone (please complete the following information):

N/A

### PR DESCRIPTION
**Pull request by FangedParakeet**
Fixes #6322 

#### Changes proposed in this Pull Request

# E2E Errors Begone

The two tests "WC - latest | upe - shopper" and "WC - latest | UPESplit - shopper" keep failing infrequently but frequently enough on GH actions for WCPay PRs. These changes make a few modifications to amend this distressing and rather annoying situation. 

Firstly SEPA enablement is removed for split-UPE E2E tests, since the split-UPE feature is not yet fully compatible with the SEPA payment method. However, we do need to enable _some_ additional LPMs on all split-UPE tests, because doing so automatically enables the multi-currency feature and adds EUR as a presentment currency, which later tests are dependent on. 

Secondly, there is a click action on a fraud protection tools tour button that is triggered inconsistently. I made this action a little more precise by using the specific class mentioned in WooCommerce's own E2E tests and adding a more robust function to ensure the button is always located and definitively clicked. 

#### Testing instructions

1. You are welcome to pull these tests locally, configure your own local E2E testing setup, and run these tests; but a more effective and accurate method would be to keep refreshing the testing jobs on this PR and ensure that they always come back fully green.
2. In either case, the tests should now always pass.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6330` or branch name `fix/6322-update-upe-e2e-tests` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6322-update-upe-e2e-tests) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: c595914e1d49d00f8d79ccbbeaa3ecf463ed3c5f
- Build time: 2023-05-24 14:02:23 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js [hunk 1]
```diff
@@ -30,8 +34,6 @@ describe( 'Enabled Split UPE', () => {
 		await shopperWCP.changeAccountCurrencyTo( 'USD' );
 		await shopperWCP.logout();
 		await merchant.login();
-		//disable SEPA
-		await merchantWCP.disablePaymentMethod( [ sepaPaymentMethod ] );
```
Comment (jbordonado) [inline tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js:34]: I think we still need this step to clean up the side effects of the test. Any specific reason to remove this?

File: tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js [hunk 2]
```diff
@@ -11,16 +11,20 @@ import { fillCardDetails, setupProductCheckout } from '../../../utils/payments';
 const { shopper, merchant, uiUnblocked } = require( '@woocommerce/e2e-utils' );
 
 const MIN_WAIT_TIME_BETWEEN_PAYMENT_METHODS = 20000;
-
-const sepaPaymentMethod = '#inspector-checkbox-control-8';
+const upeMethodCheckboxes = [
```
Comment (jbordonado) [inline tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js]:
```suggestion
const UPE_METHOD_CHECKBOXES = [
```

**Review by jbordonado — COMMENTED** [review]
I also noticed the flakiness. Thank you for working on it.

Left 2 minor comments as I was passing by :)

**Review by timur27 — APPROVED** [review]
I've re-run the `E2E Tests - Pull Request` four times ([1](https://github.com/Automattic/woocommerce-payments/actions/runs/5046280100), [2](https://github.com/Automattic/woocommerce-payments/actions/runs/5046508388), [3](https://github.com/Automattic/woocommerce-payments/actions/runs/5045926126), [4](https://github.com/Automattic/woocommerce-payments/actions/runs/5046046968)) and no failed tests were observed. I didn't run `E2E Tests - All` because they are not running the test groups which we are interested in. Locally tests failed, but for some other reason than in the GitHub Actions error logs which are linked to the issue.

Thanks for updating the tests to enable/disable payment methods other than SEPA. I think it's a great time to make this change happen. At the same time, my hunch is that the root cause of the failures was in `tests/e2e/utils/flows.js` which you updated as well. Something strange, as these tests were passing all the time previously and started to fail unexpectedly. 

I agree with Jordan that we need to disable the previously enabled payment methods on the test finish, just to clean up the setup state for the next tests running. I'm looking forward for this change to be applied, but in the meantime I'm giving this PR a green light, as there was nothing else I spotted.

File: tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js (hunk 1 above)
Comment (FangedParakeet) [reply tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js:34]: Nice catch, caught me slippin'. Amended in be2f198194db25d1666038ca3ba83127f3cbc256. 👍

File: tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js (hunk 2 above)
Comment (FangedParakeet) [reply tests/e2e/specs/upe-split/shopper/shopper-upe-enabled-all-flows.spec.js]: Good shout to correct the convention here: amended in 017e819c94a77dcbead5c94b7f758eb63e8e353b.

**Review by jbordonado — APPROVED** [review]

**Pull request merged**